mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/plan/synchronization/impl/ConfiguredAutomaticIndexingSynchronizationStrategy.java

```diff
@@ -4,38 +4,37 @@
  * License: GNU Lesser General Public License (LGPL), version 2.1 or later
  * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
  */
-package org.hibernate.search.mapper.orm.automaticindexing.session.impl;
+package org.hibernate.search.mapper.pojo.plan.synchronization.impl;
+
 
 import java.util.concurrent.CompletableFuture;
 import java.util.function.Consumer;
 
 import org.hibernate.search.engine.backend.common.spi.EntityReferenceFactory;
+import org.hibernate.search.engine.backend.common.spi.MultiEntityOperationExecutionReport;
 import org.hibernate.search.engine.backend.work.execution.DocumentCommitStrategy;
 import org.hibernate.search.engine.backend.work.execution.DocumentRefreshStrategy;
 import org.hibernate.search.engine.backend.work.execution.OperationSubmitter;
 import org.hibernate.search.engine.reporting.FailureHandler;
-import org.hibernate.search.mapper.orm.automaticindexing.session.AutomaticIndexingSynchronizationConfigurationContext;
-import org.hibernate.search.mapper.orm.common.EntityReference;
-import org.hibernate.search.mapper.orm.work.SearchIndexingPlanExecutionReport;
-import org.hibernate.search.mapper.orm.work.impl.SearchIndexingPlanExecutionReportImpl;
+import org.hibernate.search.mapper.pojo.plan.synchronization.IndexingPlanSynchronizationStrategyConfigurationContext;
+import org.hibernate.search.mapper.pojo.work.SearchIndexingPlanExecutionReport;
 import org.hibernate.search.mapper.pojo.work.spi.PojoIndexingPlan;
 import org.hibernate.search.util.common.impl.Contracts;
 
-public final class ConfiguredAutomaticIndexingSynchronizationStrategy {
+public abstract class ConfiguredAutomaticIndexingSynchronizationStrategy<E, R extends SearchIndexingPlanExecutionReport<E>> {
```

Comment:
Is there any point in making this abstract? As far as I can see, all implementations are the same.

Reply:
IIRC this abstract was for the:
```
-						.thenApply( SearchIndexingPlanExecutionReportImpl::new );
+						.thenApply( this::createReport );
```
as reports were mapper specific. (at least originally 😄)